jdaviz/core/marks.py

```diff
@@ -257,3 +257,13 @@ def __init__(self, viewer, x=[], y=[], **kwargs):
 
 class LineAnalysisContinuumRight(LineAnalysisContinuumLeft):
     pass
+
+
+class LineUncertainties(Lines):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+class ScatterMask(Scatter):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
```

Comment:
If we split out some of the logic from `BaseSpectrumVerticalLine` into an even-higher base class, then these could inherit from that and get automatic unit conversion support (although we'll need to write the y-unit logic yet).

Alternatively, these could be implemented similarly to the `Shadow` mark class and avoid the need for unit conversion entirely (by just being plotted relative to the parent spectrum object).

Reply:
Please link here if you do decide to create a follow-up ticket. Thanks!